### PR TITLE
bessctl up4 configuration is executed once

### PR DESCRIPTION
**Pull request by gatici**
# Description

Do not run bessctl up4 configuration twice in sdcore-upf-operator.
This patch creates a dummy file when bessctl configure is executed. So, bessctl configure command is executed once.
Fixing FileNotFoundError error while getting namespace through the file: "/var/run/secrets/kubernetes.io/serviceaccount/namespace" during the k8s service patcher creation.

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [ ] I have bumped the version of the library

### CODE REVIEW

File: tests/unit/test_charm.py [hunk 1]
```diff
@@ -241,11 +241,54 @@ def test_given_iptables_rule_is_created_when_bessd_pebble_ready_then_rule_is_not
     @patch("ops.model.Container.exec")
     @patch("ops.model.Container.pull", new=Mock)
     @patch("ops.model.Container.exists")
-    def test_given_can_connect_to_bessd_when_bessd_pebble_ready_then_bessctl_configure_is_executed(
+    def test_given_can_connect_to_bessd_when_bessd_pebble_ready__bessctl_configure_is_executed(
```
Comment (ghislainbourgeois) [inline tests/unit/test_charm.py:244]:
```suggestion
    def test_given_can_connect_to_bessd_when_bessd_pebble_ready_then_bessctl_configure_is_executed(
```

File: tests/unit/test_charm.py [hunk 2]
```diff
@@ -241,11 +241,54 @@ def test_given_iptables_rule_is_created_when_bessd_pebble_ready_then_rule_is_not
...
+            timeout=30,
+            environment={"CONF_FILE": "/etc/bess/conf/upf.json", "PYTHONPATH": "/opt/bess"},
+        )
+
+    @patch("lib.charms.kubernetes_charm_libraries.v0.multus.KubernetesMultusCharmLib.is_ready")
+    @patch("ops.model.Container.push", new=Mock)
+    @patch("ops.model.Container.exec")
+    @patch("ops.model.Container.pull", new=Mock)
+    @patch("ops.model.Container.exists")
+    def test_bessd__pebble_ready__bessctl_executed_file_exists__bessctl_configure_not_executed(
```
Comment (ghislainbourgeois) [inline tests/unit/test_charm.py:264]:
```suggestion
    def test_given_can_connect_and_bessctl_executed_file_exists_then_bessctl_configure_not_executed(
```

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -241,11 +241,54 @@ def test_given_iptables_rule_is_created_when_bessd_pebble_ready_then_rule_is_not
...
+            )
+            not in patch_exec.mock_calls
+        )
+
+    @patch("charms.kubernetes_charm_libraries.v0.multus.KubernetesMultusCharmLib.is_ready")
+    @patch("ops.model.Container.push", new=Mock)
+    @patch("ops.model.Container.exec")
+    @patch("ops.model.Container.pull", new=Mock)
+    @patch("ops.model.Container.exists")
+    def test_bessd__pebble_ready__bessctl_executed_file_dont_exist__bessctl_configure_executed(
```
Comment (ghislainbourgeois) [inline tests/unit/test_charm.py:287]:
```suggestion
    def test_given_can_connect_and_bessctl_executed_file_dont_exist_then_bessctl_configure_executed(
```

**Review by ghislainbourgeois — CHANGES_REQUESTED** [review]
Small fixes to the test name, otherwise LGTM

File: tests/unit/test_charm.py (hunk 1 above)
Comment (gatici) [reply tests/unit/test_charm.py]: Done

File: tests/unit/test_charm.py (hunk 2 above)
Comment (gatici) [reply tests/unit/test_charm.py]: Done

File: tests/unit/test_charm.py (hunk 3 above)
Comment (gatici) [reply tests/unit/test_charm.py]: Done

**Comment by gatici** [comment]
Ready to review again.

**Review by ghislainbourgeois — APPROVED** [review]

File: src/charm.py [hunk 4]
```diff
@@ -94,6 +89,18 @@ def __init__(self, *args):
...
+    def _get_k8s_service_patcher(self) -> KubernetesServicePatch:
+        try:
+            return KubernetesServicePatch(
+                charm=self,
+                ports=[
+                    ServicePort(name="pfcp", port=8805, protocol="UDP"),
+                    ServicePort(name="prometheus-exporter", port=PROMETHEUS_PORT),
+                ],
+            )
+        except FileNotFoundError as e:
```
[the remark this answers is not in the document]
Comment (gatici) [reply src/charm.py:101]: This error is happening everytime if we do not have that file "/var/run/secrets/kubernetes.io/serviceaccount/namespace". Hence, we usually get FileNotFoundError if microk8s is used in our environment. The error is not directly related with the unit tests. When we instantiate an instance from UPFOperatorCharm, object has a service_patcher using KubernetesServicePatch class. KubernetesServicePatch class has namespace property which looks for indicated file which causes FileNotFoundError in some environments. I think, patching KubernetesServicePatch class in observability_libs could be more stable solution.

**Pull request merged**